Lib/typing.py

```diff
@@ -1281,10 +1281,8 @@ def _determine_new_args(self, args):
         # anything more exotic than a plain `TypeVar`, we need to consider
         # edge cases.
 
-        if any(isinstance(p, TypeVarTuple) for p in self.__parameters__):
-            raise NotImplementedError(
-                "Type substitution for TypeVarTuples is not yet implemented"
-            )
+        if len(self.__parameters__) == 1 and isinstance(self.__parameters__[0], TypeVarTuple):
```

Comment:
I'm not sure it's sufficient to only check for the case where `len(self.__parameters__) == 1` - we could have e.g.:

```python
T1 = TypeVar('T1')
T2 = TypeVar('T2')
Ts = TypeVarTuple('Ts')
class A(Generic[T1, T2, Unpack[Ts]]): pass
B = A[int, T2, Unpack[Ts]]
C = B[str, float]
print(C)
```

There, `B.__parameters__` is `(T2, Ts)`, causing a `TypeError` later on in the code.

Reply:
You are right. The previous code only worked with existing tests. New code adds tests for generics with multiple type variables.